### PR TITLE
Contacts team carousel

### PR DESCRIPTION
**Pull request by MihneaDinuGL**

### CODE REVIEW

**Review by jab-khanGL — APPROVED** [review]
Looks good and slick! The edit body does look interesting.

File: src/components/case-study-single/EditBody.jsx [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+  const [imageIndex, setImageIndex] = useState();
+  const [responseStatus, setResponseStatus] = useState(null);
+  const [downloadUri, setDownloadUri] = useState();
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (imageIndex !== undefined && imageIndex !== null && downloadUri) {
+      let newUpdatedBody = [...updatedBody];
```
Comment (sourcesmith) [inline src/components/case-study-single/EditBody.jsx]: const reference

File: src/components/case-study-single/EditBody.jsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+      newUpdatedBody[imageIndex].imageId = downloadUri;
+      setUpdatedBody(newUpdatedBody);
+      let newChangeHistory = [...changeHistory];
```
Comment (sourcesmith) [inline src/components/case-study-single/EditBody.jsx]: const reference.

File: src/components/case-study-single/EditBody.jsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+      overview: updatedOverview,
+      body: body,
+      coverImageId: coverImageId,
+      spotlight: spotlight,
+    };
+    put(id, updatedCaseStudy, setUpdateStatus, setPageData);
+  };
+
+  const addRow = () => {
+    let newUpdatedBody = [...updatedBody];
```
Comment (sourcesmith) [inline src/components/case-study-single/EditBody.jsx]: const

File: src/components/case-study-single/EditBody.jsx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    newUpdatedBody.push({
+      imageLink: '',
+      markdownText: '',
+    });
+    setUpdatedBody(newUpdatedBody);
+    let newChangeHistory = [...changeHistory];
```
Comment (sourcesmith) [inline src/components/case-study-single/EditBody.jsx]: const

File: src/components/case-study-single/EditBody.jsx [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+  const handleUndo = () => {
+    setHistoryPointer(historyPointer > 0 && historyPointer - 1);
+  };
+
+  const handleRedo = () => {
+    setHistoryPointer(historyPointer < changeHistory.length - 1 && historyPointer + 1);
+  };
+
+  const deleteRedoHistory = () => {
+    let newChangeHistory = [...changeHistory];
```
Comment (sourcesmith) [inline src/components/case-study-single/EditBody.jsx]: const

File: src/components/case-study-single/EditBody.jsx [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+    newChangeHistory.splice(historyPointer + 1, changeHistory.length - historyPointer);
+    console.log('newChangeHistory', newChangeHistory);
+  };
+
+  const handleChangeTextarea = (value, index) => {
+    let newChangeHistory = [...changeHistory];
```
Comment (sourcesmith) [inline src/components/case-study-single/EditBody.jsx]: const

File: src/components/case-study-single/EditBody.jsx [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+  const handleChangeRowImage = (file, index) => {
+    uploadAttachment(file, setResponseStatus, setDownloadUri);
+    setImageIndex(index);
+  };
+
+  const handleChangeCoverImage = (file) => {
+    uploadAttachment(file, setResponseStatus, setDownloadUri);
+    setUpdatedCoverImageLink(downloadUri);
+    let newPageData = pageData;
```
Comment (sourcesmith) [inline src/components/case-study-single/EditBody.jsx]: const

**Review by sourcesmith — COMMENTED** [review]
I think there maybe some confusion over the meaning of const.  Const means a reference to an object cannot be reassigned (or a primitive value changed) but does not stop the referenced object itself from being mutated.

Otherwise, it is fine.

File: src/components/case-study-single/EditBody.jsx (hunk 7 above)
Comment (MihneaDinuGL) [reply src/components/case-study-single/EditBody.jsx]: Hello, this is actually related to Tom's case studies single page. However, I talked to him and he confirmed he'd be happy for me to make the changes, so everything has been updated and pushed

**Review by sourcesmith — APPROVED** [review]

**Pull request merged**